File: lib/models/ScanResult.dart
import 'package:doc_scan/models/TextObservation.dart';

class ScanResult {
  final VisionResult visionResult;
  final String originalImageBase64;
  final String croppedImageBase64;

  ScanResult(this.visionResult, this.originalImageBase64, this.croppedImageBase64);

factory ScanResult.fromJson(Map<String, dynamic> json){
    return ScanResult(
      VisionResult.fromJson(json['visionResult']),
      json['originalImageBase64'],
      json['croppedImageBase64']
    );
  }
}

File: android/app/src/main/java/com/example/doc_scan/model/ScanResult.java
package com.example.doc_scan.model;

public class ScanResult {
    public VisionResult visionResult;
    public String originalImageBase64;
    public String croppedImageBase64;

    public ScanResult(VisionResult visionResults, String originalImageBase64, String croppedImageBase64) {
        this.visionResult = visionResults;
        this.originalImageBase64 = originalImageBase64;
        this.croppedImageBase64 = croppedImageBase64;
    }
}
